### PR TITLE
handle errors

### PR DESCRIPTION
**Pull request by bunsenstraat**

### CODE REVIEW

**Review by yann300 — APPROVED** [review]

File: libs/remixd/src/bin/remixd.ts [hunk 1]
```diff
@@ -36,12 +36,21 @@ const ports = {
 }
 
 const killCallBack: Array<Function> = []
-function startService<S extends 'git' | 'hardhat' | 'folder'> (service: S, callback: (ws: WS, sharedFolderClient: servicesList.Sharedfolder) => void) {
+function startService<S extends 'git' | 'hardhat' | 'folder'> (service: S, callback: (ws: WS, sharedFolderClient: servicesList.Sharedfolder, error:Error) => void) {
```
Comment (yann300) [inline libs/remixd/src/bin/remixd.ts:39]: should this be an optional argument?

File: libs/remixd/src/websocket.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ export default class WebSocket {
 
   constructor (public port: number, public opt: WebsocketOpt, public getclient: () => ServiceClient) {} //eslint-disable-line
 
-  start (callback?: (ws: WS, client: ServiceClient) => void): void {
+  start (callback?: (ws: WS, client: ServiceClient, errror: Error) => void): void {
```
Comment (yann300) [inline libs/remixd/src/websocket.ts:12]: typo

File: libs/remixd/src/bin/remixd.ts (hunk 1 above)
Comment (bunsenstraat) [reply libs/remixd/src/bin/remixd.ts:39]: sure

File: libs/remixd/src/websocket.ts (hunk 2 above)
Comment (bunsenstraat) [reply libs/remixd/src/websocket.ts:12]: :D sticky keys. will fix.

File: libs/remixd/src/bin/remixd.ts [hunk 3]
```diff
@@ -76,7 +85,8 @@ function startService<S extends 'git' | 'hardhat' | 'folder'> (service: S, callb
     console.log('\x1b[33m%s\x1b[0m', '[WARN] Any application that runs on your computer can potentially read from and write to all files in the directory.')
     console.log('\x1b[33m%s\x1b[0m', '[WARN] Symbolic links are not forwarded to Remix IDE\n')
     try {
-      startService('folder', (ws: WS, sharedFolderClient: servicesList.Sharedfolder) => {
+      startService('folder', (ws: WS, sharedFolderClient: servicesList.Sharedfolder, error: any) => {
+        if (error) errorHandler(error, 'folder')
```
Comment (yann300) [inline libs/remixd/src/bin/remixd.ts:89]: if `error` you might want to return here so the rest is not executed.

File: libs/remixd/src/bin/remixd.ts [hunk 4]
```diff
@@ -85,7 +95,8 @@ function startService<S extends 'git' | 'hardhat' | 'folder'> (service: S, callb
       const hardhatConfigFilePath = absolutePath('./', program.sharedFolder) + '/hardhat.config.js'
       const isHardhatProject = fs.existsSync(hardhatConfigFilePath)
       if (isHardhatProject) {
-        startService('hardhat', (ws: WS, sharedFolderClient: servicesList.Sharedfolder) => {
+        startService('hardhat', (ws: WS, sharedFolderClient: servicesList.Sharedfolder, error: Error) => {
+          if (error) errorHandler(error, 'hardhat')
```
Comment (yann300) [inline libs/remixd/src/bin/remixd.ts:99]: same as above

**Review by yann300 — APPROVED** [review]

**Pull request merged**